src/hal/mod.rs

```diff
@@ -22,14 +22,23 @@ and each such struct has a `setup()` method that configures the hardware
 */
 
 
+#[cfg(feature = "mcu_lpc17xx")]
```

Comment:
But wouldn't you then just test with the features enabled (which we already do)? As a bonus it has exposed some misuse of hal code.